pkgs/development/python-modules/pypdfium2/default.nix

```diff
@@ -80,43 +58,25 @@ buildPythonPackage rec {
     pdfium-binaries
   ];
 
-  # Build system insists on fetching from the internet unless "cached" files
-  # are prepared. Even then, some code patching needs to happen to make it not
-  # talk to the internet.
+  preBuild = ''
+    getVersion() {
+      cat ${pdfium-binaries}/VERSION | grep $1 | sed 's/.*=//'
+    }
+    export GIVEN_FULLVER="$(getVersion MAJOR).$(getVersion MINOR).$(getVersion BUILD).$(getVersion PATCH)"
+  '';
 
-  # The project doesn't seem very open to allow for offline building either,
-  # see: https://github.com/pypdfium2-team/pypdfium2/discussions/274
-  preBuild =
+  env =
     let
-      pdfiumLib = lib.makeLibraryPath [ pdfium-binaries ];
-      inputVersionFile = (pkgs.formats.json { }).generate "version.json" {
-        version = lib.strings.toInt pdfiumVersion;
-        source = "generated";
-        flags = [ ];
-        run_lds = [ pdfiumLib ];
-        guard_symbols = false;
-      };
-      bindingsDir = "data/bindings";
-      headersDir = "${bindingsDir}/headers";
-      versionFile = "${bindingsDir}/version.json";
+      ldPath = lib.makeLibraryPath [ pdfium-binaries ];
     in
-    ''
-      # Preseed the headers and version file
-      mkdir -p ${bindingsDir}
-      cp -r ${headers}/public ${headersDir}
-      install -m 644 ${inputVersionFile} ${versionFile}
-
-      # Make generated bindings consider pdfium derivation path when loading dynamic libraries
-      substituteInPlace setupsrc/pypdfium2_setup/emplace.py \
-        --replace-fail 'build_pdfium_bindings(pdfium_ver, flags=flags, guard_symbols=True, run_lds=[])' \
-                       'build_pdfium_bindings(pdfium_ver, flags=flags, guard_symbols=True, run_lds=["${pdfiumLib}"])'
-
-      # Short circuit the version pull from the internet
-      substituteInPlace setupsrc/pypdfium2_setup/packaging_base.py \
-        --replace-fail 'PdfiumVer.to_full(build)._asdict()' \
-                       '{"major": 133, "minor": 0, "build": ${pdfiumVersion}, "patch": 1}'
-    '';
-  env.PDFIUM_PLATFORM = "system:${pdfiumVersion}";
+    {
+      PDFIUM_PLATFORM = "system-search:${pdfium-binaries.version}";
+      PDFIUM_HEADERS = "${pdfium-binaries}/include";
+      PDFIUM_BINARY = "${pdfium-binaries}/lib/libpdfium${stdenv.targetPlatform.extensions.sharedLibrary}";
+      LD_LIBRARY_PATH = ldPath;
+      DYLD_LIBRARY_PATH = ldPath;
```

Comment:
As you're setting the path with `PDFIUM_BINARY`, I think you don't need to add it to the OS library search path as well?